src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -70,14 +71,33 @@ desugarModule
   => Module
   -> Desugar m Module
 desugarModule (Module ss coms name decls (Just exps)) = do
-  (newExpss, declss) <- unzip <$> parU (sortBy classesFirst decls) (desugarDecl name exps)
-  return $ Module ss coms name (concat declss) $ Just (exps ++ catMaybes newExpss)
+  let classDecls = filter isTypeClassDeclaration decls
```

Comment:
Probably better to use [`partition`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-List.html#v:partition) here.